runtime/resolvers/sql.go

```diff
@@ -0,0 +1,263 @@
+package resolvers
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"time"
+
+	runtimev1 "github.com/rilldata/rill/proto/gen/rill/runtime/v1"
+	"github.com/rilldata/rill/runtime"
+	compilerv1 "github.com/rilldata/rill/runtime/compilers/rillv1"
+	"github.com/rilldata/rill/runtime/drivers"
+	"github.com/rilldata/rill/runtime/pkg/duckdbsql"
+	"github.com/rilldata/rill/runtime/queries"
+	"github.com/rilldata/rill/runtime/reconcilers"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+func init() {
+	runtime.RegisterAPIResolverInitializer("SQL", New)
+}
+
+type SQLResolver struct {
+	resolvedSQL string
+	deps        []*runtimev1.ResourceName
+	olap        drivers.OLAPStore
+	releaseFunc func()
+}
+
+func New(ctx context.Context, opts *runtime.APIResolverOptions) (runtime.APIResolver, error) {
```

Comment:
nit: Seems like it doesn't need to be exported. And since we'll be adding more resolvers, should probably be something like `newSQL`.